sdk/src/androidTest/java/ch/admin/bag/covidcertificate/eval/chain/TimestampServiceTest.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021 Ubique Innovation AG <https://www.ubique.ch>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package ch.admin.bag.covidcertificate.eval.chain
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.admin.bag.covidcertificate.eval.data.Eudgc
+import ch.admin.bag.covidcertificate.eval.models.Bagdgc
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.Instant
+
+@RunWith(AndroidJUnit4::class)
+class TimestampServiceTest {
+
+	// RECALL: The service returns null if there was NO error
+
+	@Test
+	fun future_expiration() {
+		val dcc = Bagdgc(Eudgc(), "", expirationTime = Instant.now().plusSeconds(60))
+		assertNull(TimestampService.decode(dcc))
+	}
+
+	@Test
+	fun past_expiration() {
+		val dcc = Bagdgc(Eudgc(), "", expirationTime = Instant.now().minusSeconds(60))
+		assertNotNull(TimestampService.decode(dcc))
```

Comment:
Should you not just check for a non-null return value but also the right error code?